File: javascript/roman-numerals/roman-numerals.js
const roman = [
  [1000, 'M'],
  [900, 'CM'],
  [500, 'D'],
  [400, 'CD'],
  [100, 'C'],
  [90, 'XC'],
  [50, 'L'],
  [40, 'XL'],
  [10, 'X'],
  [9, 'IX'],
  [5, 'V'],
  [4, 'IV'],
  [1, 'I'],
];

export const toRoman = (num) => {
  let res = '';

  roman.forEach((r) => {
    const [value, letter] = r;
    while (num >= value) {
      res += letter;
      num -= value;
    }
  });

  return res;
};

File: csharp/roman-numerals/RomanNumerals.cs
using System;
using System.Text;
using System.Collections.Generic;

public static class RomanNumeralExtension
{
    static List<(string, int)> map = new List<(string, int)>(){
        ("M", 1000), ("CM", 900), ("D", 500), ("CD", 400),
        ("C", 100), ("XC", 90), ("L", 50), ("XL", 40),
        ("X", 10), ("IX", 9), ("V", 5), ("IV", 4), ("I", 1),
    };

    public static string ToRoman(this int value)
    {
        StringBuilder s = new StringBuilder();

        foreach(var entry in map) {
            while (value >= entry.Item2) {
                s.Append(entry.Item1);
                value -= entry.Item2;
            }
        }

        return s.ToString();
    }
}